tests/e2e/cluster_upgrade_test.go

```diff
@@ -0,0 +1,107 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	CI "github.com/terraform-redhat/terraform-provider-rhcs/tests/ci"
+	CMS "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/cms"
+	CON "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/constants"
+	EXE "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/exec"
+	"github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/openshift"
+)
+
+var _ = Describe("RHCS Provider Test", func() {
+	Describe("Upgrade cluster tests", func() {
+
+		var targetV string
+		var clusterID string
+		BeforeEach(OncePerOrdered, func() {
+			clusterID, err = CI.PrepareRHCSClusterByProfileENV()
+			Expect(err).ToNot(HaveOccurred())
+
+		})
+		Context("Author:amalykhi-Critical-OCP-63153 @OCP-63153 @amalykhi", func() {
+			It("Update a ROSA STS cluster with RHCS provider", CI.Update,
```

Comment:
Missing checkpoint of "downgrade" cluster to lower version.
Is there any cluster status on the terraform resource state when the cluster upgrade is triggered,like 'Scheduled/Upgrading' status? 
check the upgraded version on the terraform resource state, not only via CMS response.